src/ethereum/frontier/vm/instructions/arithmetic.py

```diff
@@ -351,7 +351,8 @@ def signextend(evm: Evm) -> None:
 
         # Now among the obtained value bytes, consider only
         # N `least significant bytes`, where N is `byte_num + 1`.
-        value_bytes = value_bytes[len(value_bytes) - 1 - byte_num :]
+        if len(value_bytes) > byte_num:
+            value_bytes = value_bytes[len(value_bytes) - 1 - byte_num :]
```

Comment:
cc @Bhargavasomu. I didn't really go too deep to understand the logic but this was breaking a test and I realized that I could fix this with a simple if. 
Happy to open a separate PR for this.

Reply:
https://github.com/ethereum/eth1.0-specs/issues/283

Reply:
I believe that `len(value_bytes) - 1 - byte_num` is overflowing U256. A simple fix for this would be `len(value_bytes) - 1 - Uint(byte_num)`.

I don't think converting to bytes32 is needed unless it adds more readability.